Sources/Amplitude/Plugins/iOS/UIKitUserInteractions.swift

```diff
@@ -0,0 +1,153 @@
+        else { return sendActionResult }
+
+        if let textField = view as? UITextField, !textField.shouldTrack(action, for: event) {
+            return sendActionResult
+        } else {
+            #if !os(tvOS)
+            if let slider = view as? UISlider, !slider.shouldTrack(action, for: event) {
+                return sendActionResult
+            }
+            #endif
+        }
+
+        let viewData = view.extractData(with: action, in: keyWindow)
+
+        UIKitUserInteractions.amplitudeInstances.allObjects.forEach {
+            $0.track(event: UserInteractionEvent(
+                viewController: viewData.viewController,
+                title: viewData.title,
+                accessibilityLabel: viewData.accessibilityLabel,
+                actionMethod: viewData.actionMethod,
+                targetViewClass: viewData.targetViewClass,
+                targetText: viewData.targetText,
+                hierarchy: viewData.hierarchy))
+        }
+
+        return sendActionResult
+    }
+}
+
+extension UIView {
+    private static let viewHierarchyDelimiter = " -> "
+
+    struct ViewData {
+        let viewController: String?
+        let title: String?
+        let accessibilityLabel: String?
+        let actionMethod: String
+        let targetViewClass: String
+        let targetText: String?
+        let hierarchy: String
+    }
+
+    func extractData(with action: Selector, in window: UIWindow) -> ViewData {
+        var targetText: String?
+
+        if let button = self as? UIButton {
+            targetText = button.titleLabel?.text
+        } else if let textField = self as? UITextField {
+            targetText = textField.placeholder
+        }
+
+        let viewController = window.rootViewController
+        let viewControllerClassName = viewController?.descriptiveTypeName
+        let viewControllerTitle = viewController?.title
+        let targetAccessibilityLabel = self.accessibilityLabel
+        let actionName = NSStringFromSelector(action)
+        let targetViewClassName = self.descriptiveTypeName
+        let viewHierarchy = sequence(first: self, next: { $0.superview })
+            .map { $0.descriptiveTypeName }
+            .joined(separator: UIView.viewHierarchyDelimiter)
+
+        return ViewData(
+            viewController: viewControllerClassName,
+            title: viewControllerTitle,
+            accessibilityLabel: targetAccessibilityLabel,
+            actionMethod: actionName,
+            targetViewClass: targetViewClassName,
+            targetText: targetText,
+            hierarchy: viewHierarchy)
+    }
+}
+
+extension UIResponder {
+    var descriptiveTypeName: String {
+        String(describing: type(of: self))
+    }
+}
+
+extension UITextField {
+    func shouldTrack(_ action: Selector, for event: UIEvent?) -> Bool {
+        // primaryActionTriggered: is triggered when the text field loses focus.
+        action == Selector(("primaryActionTriggered:"))
```

Comment:
I'd expect this to fire at roughly the same points as the delegate method `func textFieldDidEndEditing(_ textField: UITextField)`, some initial searches sound like it corresponds to pressing the return key - does this behave the same way?

Reply:
That's correct, `primaryActionTriggered:` is only triggered when the return key is pressed. We may want to capture text field interactions in a totally different way as no action method is triggered when a text field loses focus.

Reply:
I'd love to get something equivalent to web's focus/blur. This still may be interesting to track, but it will generally coincide with the blur / didEndEditing event.